// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  // test exchange 3 list of coords
  rpc TranslateCoordList3 (TranslateCoordList3Request) returns (CoordList3) {}
  // test exchange list of coord3Ds
  rpc TranslateCoord3List (TranslateCoord3ListRequest) returns (Coord3List) {}
  // test exchange list of coord3Ds with streaming
  rpc TranslateCoord3Stream (stream TranslateCoord3Request) returns (stream Coord3) {}
}

// 3 list of coords
message CoordList3 {
    repeated double x = 1;
    repeated double y = 2;
    repeated double z = 3;
}

// translate 3 list of coords
message TranslateCoordList3Request {
    CoordList3 coord_list = 1;
    double tx = 2;
    double ty = 3;
    double tz = 4;
}

// translate 3 list of coords
message TranslateCoord3Request {
    Coord3 coords = 1;
    double tx = 2;
    double ty = 3;
    double tz = 4;
}

// coord3D
message Coord3 {
    double x = 1;
    double y = 2;
    double z = 3;
}

// list of coord3D
message Coord3List {
    repeated Coord3 coords = 1;
}

// translate list of coord3Ds
message TranslateCoord3ListRequest {
    Coord3List coord_list= 1;
    double tx = 2;
    double ty = 3;
    double tz = 4;
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
